src/middleware/extract.company.number.middleware.ts

```diff
@@ -0,0 +1,42 @@
+import { Request, Response, NextFunction } from "express";
+import { checkCompanyNumberFormatIsValidate } from "../utils/format/company.number.format";
+import { ContextKeys } from "../utils/constants/context.keys";
+import { getCompanyNumber } from "../utils/session";
+import { logger } from "../utils/logger";
+import { ValidateCompanyNumberFormat } from "../utils/validate/validate.company.number";
+
+declare module 'express-serve-static-core' {
+    interface Request {
+        companyNumber?: string;
+    }
+}
+
+export const extractCompanyNumberMiddleware = (req: Request, res: Response, next: NextFunction) => {
+    try {
```

Comment:
You shouldn't cover large parts of the code with a try catch that doesn't rethrow an error, because if there is a bug it will get caught instead of breaking.